Sia.Tests/Auxiliary/ContextTests.cs

```diff
@@ -0,0 +1,78 @@
+namespace Sia.Tests.Auxiliary;
+
+public class ContextTests
+{
+    [Fact]
+    public async Task ContextValue_Assign_Test()
+    {
+        // Arrange
+        const string valueForThread1 = "ThreadValue1";
+        const string valueForThread2 = "ThreadValue2";
+        var result1 = string.Empty;
+        var result2 = string.Empty;
+
+        Context<string>.Current = "The best Sia";
+
+        var task1 = Task.Run(() =>
+        {
+            Context<string>.Current = valueForThread1;
+            result1 = Context.Get<string>();
+        });
+
+        var task2 = Task.Run(() =>
+        {
+            Context<string>.Current = valueForThread2;
+            result2 = Context.Get<string>();
+        });
+
+        // Act
+        await Task.WhenAll(task1, task2);
+
+        // Assert
+        Assert.Throws<NotSupportedException>(Context.Get<string>); // TODO: Assign value in thread will override the data in main thread.
```

Comment:
@sicusa Although I'm not sure if it was designed this way on purpose, I feel like something is not quite right.